test/unit/mdc-tabs/mdc-tab-bar.test.js

```diff
@@ -0,0 +1,245 @@
+  const tabBarWidth =
+    component.getDefaultFoundation().adapter_.getOffsetWidth();
+
+  assert.equal(tabBarWidth, root.offsetWidth);
+});
+
+test('adapter#setStyleForIndicator sets a given property to the given value', () => {
+  const {indicator, component} = setupTest();
+
+  component.getDefaultFoundation().adapter_.setStyleForIndicator('width', '200px');
+  assert.equal(indicator.style.width, '200px');
+});
+
+test('adapter#getOffsetWidthForIndicator returns the width of the active tab indicator', () => {
+  const {indicator, component} = setupTest();
+  const indicatorWidth =
+    component.getDefaultFoundation().adapter_.getOffsetWidthForIndicator();
+
+  assert.equal(indicatorWidth, indicator.offsetWidth);
+});
+
+test('adapter#notifyChange emits MDCTabBar:change with event data', () => {
+  const {component} = setupTest();
+  const data = td.func('eventDataObj');
+
+  const handler = (d) => {
+    assert.equal(data, d);
+  };
+
+  window.addEventListener('MDCTabBar:change', () => {
+    handler(data);
+  });
+  component.getDefaultFoundation().adapter_.notifyChange(data);
+});
+
+test('adapter#getNumberOfTabs returns the number of tabs', () => {
+  const {component} = setupTest();
+
+  assert.equal(component.getDefaultFoundation().adapter_.getNumberOfTabs(), 3);
+});
+
+test('adapter#isTabActiveAtIndex returns true if index equals activeTab index', () => {
+  const {component} = setupTest();
+  const targetIndex = 1;
+
+  component.tabs[targetIndex].isActive = true;
+
+  assert.isTrue(component.getDefaultFoundation().adapter_.isTabActiveAtIndex(targetIndex));
+});
+
+test('adapter#setTabActiveAtIndex sets tab at target index to true or false', () => {
+  const {component} = setupTest();
+  const targetIndex = 0;
+
+  component.getDefaultFoundation().adapter_.setTabActiveAtIndex(targetIndex, true);
+
+  assert.isTrue(component.tabs[targetIndex].foundation_.isActive());
+});
+
+test('adapter#isDefaultPreventedOnClickForTabAtIndex returns the value ' +
+  ' of preventsDefaultOnClick', () => {
+  const {component} = setupTest();
+  const targetIndex = 0;
+
+  component.tabs[targetIndex].preventDefaultOnClick = true;
+
+  assert.isTrue(component.getDefaultFoundation().adapter_.isDefaultPreventedOnClickForTabAtIndex(targetIndex));
+});
+
+test('adapter#setPreventDefaultOnClickForTabAtIndex sets preventDefault ' +
+  ' for tab at index', () => {
+  const {component} = setupTest();
+  const targetIndex = 0;
+
+  component.getDefaultFoundation().adapter_.setPreventDefaultOnClickForTabAtIndex(targetIndex, true);
+
+  assert.isTrue(component.tabs[targetIndex].preventDefaultOnClick);
+});
+
+test('adapter#measureTabAtIndex calls measureSelf on the tab at a given index', () => {
+  const {component} = setupTest();
+  const targetIndex = 1;
+
+  component.getDefaultFoundation().adapter_.measureTabAtIndex(targetIndex);
+
+  assert.equal(component.tabs[targetIndex].computedWidth, component.tabs[targetIndex].root_.offsetWidth);
```

Comment:
here's an example of where mocks would be useful. If you had a `MockTab` component used within the tab factory, you could make `measureSelf()` a stub and just verify that it was invoked. This is perfectly acceptable and preferred because:

- `measureSelf()` is part of `MDCTab`'s public API
- `measureSelf()` is tested within `mdc-tab.test.js`
- If `measureSelf()` has bugs, it will not affect this test (nor should it)